### PR TITLE
[AutoParallel] Converter

### PR DESCRIPTION
**Pull request by zhaoyinglia**
### PR types

New features
### PR changes

Others
### Describe

- Add Converter implement of AutoParallel.
- convert tensors' values from one parallel strategy to another one.

### CODE REVIEW

File: python/paddle/distributed/auto_parallel/converter.py [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    def cur_strategy(self, cur_strategy):
+        if not cur_strategy:
+            warnings.warn("'cur_strategy' is None, "
+                          "there are not parameter in cur process")
+        if not isinstance(cur_strategy, dict):
+            raise TypeError("The type of 'cur_strategy' should be 'dict', "
+                            "but got '{}'.".format(str(type(cur_strategy))))
+        self._cur_strategy = cur_strategy
+
+    def convert(self, strict=True):
```
Comment (JZ-LIANG) [inline python/paddle/distributed/auto_parallel/converter.py:74]: explain meaning of "strict" argument in doc

File: python/paddle/distributed/auto_parallel/converter.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        convert parameters
+
+        Returns:
+            converted parameters(dict)
+
+        Examples:
+            .. code-block:: python
+
+                import numpy as np
+                complete_param = np.arange(4).reshape([2, 2])
```
Comment (JZ-LIANG) [inline python/paddle/distributed/auto_parallel/converter.py]: complete_param --> complete_params

File: python/paddle/distributed/auto_parallel/converter.py [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+        self._logger.info("Start to convert parameters.")
+        for param_name in self._cur_strategy:
+            if param_name not in self._pre_strategy:
+                param_not_in_pre.append(param_name)
+                continue
+            if param_name not in self._params_dict:
+                param_not_in_ckpt.append(param_name)
+                continue
+
+            param_list = self._params_dict[param_name]
```
Comment (JZ-LIANG) [inline python/paddle/distributed/auto_parallel/converter.py]: "param_xxx" --> "tesnor_xxx" a more proper and meaningful naming

File: python/paddle/distributed/auto_parallel/converter.py (hunk 1 above)
Comment (zhaoyinglia) [reply python/paddle/distributed/auto_parallel/converter.py:74]: done.

File: python/paddle/distributed/auto_parallel/converter.py (hunk 2 above)
Comment (zhaoyinglia) [reply python/paddle/distributed/auto_parallel/converter.py]: done.

File: python/paddle/distributed/auto_parallel/converter.py [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+                process, complete_shape, dims_mapping, process_shape,
+                process_group)
+            index = process_group.index(process)
+            if partition_index not in merged_partiton:
+                merged_partiton.append(partition_index)
+                Converter.merge(partition_param_list, param_list[index],
+                                partition_index, complete_shape)
+
+        assert len(partition_param_list) == 1 or not partition_param_list, \
+            "Fail to merge parameter"
```
Comment (JZ-LIANG) [inline python/paddle/distributed/auto_parallel/converter.py]: also throw out the information of failed tensor, like name, pre_dist_attr, target_dist_attr

Comment (zhaoyinglia) [reply python/paddle/distributed/auto_parallel/converter.py]: done.

File: python/paddle/distributed/auto_parallel/converter.py (hunk 3 above)
Comment (zhaoyinglia) [reply python/paddle/distributed/auto_parallel/converter.py]: done.

**Review by XieYunshen — APPROVED** [review]
LGTM

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Pull request merged**